### PR TITLE
chore(BUX-587): rename pulse -> Block Headers Service

### PR DESCRIPTION
**Pull request by wregulski**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/pulse/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/pulse/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [x] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: cli/flags.go [hunk 1]
```diff
@@ -26,22 +26,22 @@ func LoadFlags(cfg *config.AppConfig) error {
 	}
 
 	cli := &cliFlags{}
-	pulseFlags := pflag.NewFlagSet("pulseFlags", pflag.ContinueOnError)
+	bhsFlags := pflag.NewFlagSet("bhsFlags", pflag.ContinueOnError)
```
Comment (dorzepowski) [inline cli/flags.go:29]: maybe `appFlags` - because `bhs` could be confusing or disrupting.

File: cmd/main.go [hunk 2]
```diff
@@ -118,7 +118,7 @@ func main() {
 
 	server.ApplyConfiguration(metrics.Register)
 
-	server.ApplyConfiguration(endpoints.SetupPulseRoutes(hs, cfg.HTTP))
+	server.ApplyConfiguration(endpoints.SetupBHSRoutes(hs, cfg.HTTP))
```
Comment (dorzepowski) [inline cmd/main.go:121]: maybe just `SetupRoutes`?

File: config/load.go [hunk 3]
```diff
@@ -97,7 +97,7 @@ func unmarshallToAppConfig(appConfig *AppConfig) error {
 }
 
 func envConfig() {
-	viper.SetEnvPrefix("pulse")
+	viper.SetEnvPrefix("bhs")
```
Comment (dorzepowski) [inline config/load.go:100]: Now I wonder why we have [declared const](https://github.com/bitcoin-sv/pulse/pull/206/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R19) with env prefix, as we are using a static string here.

**Comment by chris-4chain** [comment]
Don't forget to change the line in `CODE_STANDARDS.md` :D 

https://github.com/bitcoin-sv/pulse/blob/f70c6cefe8e1bc5504f2993116709f493ab91576/.github/CODE_STANDARDS.md?plain=1#L236

File: database/import.go [hunk 4]
```diff
@@ -61,7 +61,7 @@ func getHeadersFile(preparedDbFilePath string, log *zerolog.Logger) (*os.File, s
 		return nil, "", fmt.Errorf("file %s does not exist or is not readable", preparedDbFilePath)
 	}
 
-	const tmpHeadersFileName = "pulse-blockheaders.csv"
+	const tmpHeadersFileName = "bhs-blockheaders.csv"
```
Comment (dorzepowski) [inline database/import.go:64]: maybe just `blockheaders.tmp.csv`
or 
```suggestion
	tmpHeadersFileName := fmt.Sprintf("%d-blockheaders.csv", time.Now().Unix())
```

File: examples/ws-subscribe-to-new-headers/main.go [hunk 5]
```diff
@@ -12,7 +12,7 @@ import (
 func main() {
 
 	client := centrifuge.NewJsonClient("ws://localhost:8080/connection/websocket", centrifuge.Config{
-		// Uncomment and adjust value if pulse has authentication turned on
+		// Uncomment and adjust value if block headersservice has authentication turned on
```
Comment (dorzepowski) [inline examples/ws-subscribe-to-new-headers/main.go:15]:
```suggestion
		// Uncomment and adjust value if block headers service has authentication turned on
```

File: release/README.DOCKER.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# Pulse: Bitcoin Headers Client
+# Block Headers Service: Bitcoin Headers Client
```
Comment (dorzepowski) [inline release/README.DOCKER.md:1]:
```suggestion
# Block Headers Service
```

File: internal/tests/testbhs/test_bhs.go [hunk 7]
```diff
@@ -47,23 +48,23 @@ type TestPulse struct {
...
-	return &When{TestPulse: p}
+// When Provides test access to block headers service service operations.
+func (p *TestBHS) When() *When {
+	return &When{TestBHS: p}
 }
 
-// NewTestPulse Start pulse for testing reason.
-func NewTestPulse(t *testing.T, ops ...pulseOpt) (*TestPulse, Cleanup) {
+// NewTestBHS Start block headers service for testing reason.
+func NewTestBHS(t *testing.T, ops ...bhsOpt) (*TestBHS, Cleanup) {
```
Comment (dorzepowski) [inline internal/tests/testbhs/test_bhs.go:67]: Maybe:
```suggestion
func NewTestBlockHeaderService(t *testing.T, ops ...bhsOpt) (*TestBHS, Cleanup) {
```

File: internal/tests/testbhs/test_bhs.go [hunk 8]
```diff
@@ -31,11 +32,11 @@ type ConfigOpt func(*config.AppConfig)
...
 type RepoOpt func(*testrepository.TestRepositories)
 
-// Cleanup represents function that is used to clean up TestPulse app.
+// Cleanup represents function that is used to clean up TestBHS app.
 type Cleanup func()
 
-// TestPulse used to interact with pulse in e2e tests.
-type TestPulse struct {
+// TestBHS used to interact with block headers service in e2e tests.
+type TestBHS struct {
```
Comment (dorzepowski) [inline internal/tests/testbhs/test_bhs.go:39]: Maybe
```suggestion
type TestBlockHeaderService struct {
```

File: internal/tests/testbhs/test_bhs.go [hunk 9]
```diff
@@ -1,26 +1,27 @@
-package testpulse
+package testbhs
```
Comment (dorzepowski) [inline internal/tests/testbhs/test_bhs.go:1]: I'm not sure about the package name,
maybe `testapp` or `testserver` or 🤷  would be better

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: transports/http/endpoints/bhs_endpoints.go [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+		prefix := "/api/v1"
+		apiRouter := engine.Group(prefix, apiMiddlewares...)
+		for _, r := range routes {
+			switch r := r.(type) {
+			case router.RootEndpoints:
+				r.RegisterEndpoints(rootRouter)
+			case router.ApiEndpoints:
+				r.RegisterApiEndpoints(apiRouter, cfg)
+			default:
+				panic(errors.New("unexpected router endpoints registrar"))
```
Comment (Nazarii-4chain) [inline transports/http/endpoints/bhs_endpoints.go:55]: Typo here

**Review by kuba-4chain — COMMENTED** [review]
There is a `domain_prefix` in `metrics/naming.go`

https://github.com/bitcoin-sv/pulse/blob/f70c6cefe8e1bc5504f2993116709f493ab91576/metrics/naming.go#L9

File: cmd/main.go (hunk 2 above)
Comment (wregulski) [reply cmd/main.go]: sure, renamed

File: config/load.go (hunk 3 above)
Comment (wregulski) [reply config/load.go]: My bad, I am using const currently

File: database/import.go (hunk 4 above)
Comment (wregulski) [reply database/import.go]: done, I also updated .gitignore with wildcard schema

File: examples/ws-subscribe-to-new-headers/main.go (hunk 5 above)
Comment (wregulski) [reply examples/ws-subscribe-to-new-headers/main.go]: done

File: release/README.DOCKER.md (hunk 6 above)
Comment (wregulski) [reply release/README.DOCKER.md]: removed

File: internal/tests/testbhs/test_bhs.go (hunk 7 above)
Comment (wregulski) [reply internal/tests/testbhs/test_bhs.go]: removed

File: internal/tests/testbhs/test_bhs.go (hunk 8 above)
Comment (wregulski) [reply internal/tests/testbhs/test_bhs.go]: renamed

File: internal/tests/testbhs/test_bhs.go (hunk 9 above)
Comment (wregulski) [reply internal/tests/testbhs/test_bhs.go]: I updated it to testapp

File: examples/ws-subscribe-to-new-headers/main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (wregulski) [reply examples/ws-subscribe-to-new-headers/main.go]: done

File: transports/http/endpoints/bhs_endpoints.go (hunk 10 above)
Comment (wregulski) [reply transports/http/endpoints/bhs_endpoints.go]: It wasn't my file, just renamed it. Also I think registrar is correct word.
I changed it to registration.

**Review by kuba-4chain — DISMISSED** [review]

**Review by Nazarii-4chain — DISMISSED** [review]

File: database/import.go (hunk 4 above)
Comment (dorzepowski) [reply database/import.go]: good catch with this .gitignore 👍 🏆

File: metrics/naming.go [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
 
-const appName = "pulse"
+const appName = "block-header-service"
 
 const requestMetricBaseName = "http_request"
 const requestCounterName = requestMetricBaseName + "_total"
 const requestDurationSecName = requestMetricBaseName + "_duration_seconds"
 
-const domainPrefix = "bux_"
+const domainPrefix = "bhs_"
```
Comment (dorzepowski) [inline metrics/naming.go:9]: Worth to discuss if we don't want it consistent across SPV Wallet stack
Maybe `bsv_` would be good prefix ❓ 
WDYT @chris-4chain ?

**Review by dorzepowski — DISMISSED** [review]
Great job 🚀 
I'm giving 🍏 approve.
But please take a look at a comment about metrics name prefix

**Review by dorzepowski — APPROVED** [review]

**Pull request merged**